### ISSUE TITLE
What is the RGB version?

### ISSUE THREAD

**Issue by Zhou2019** [zhu-xlab/SSL4EO-S12#6]
Thanks for your great work!! It is awesome.

And I have some questions as follows:
1. What is the RGB version? What data does it contain?
    Is it a subset of RGB images in S2 satellite data? Or did the author synthesize RGB from multispectral images?
2. Is it convenient for the author to provide the performance metrices of the model in the second table of the [Pre-trained models](https://github.com/zhu-xlab/SSL4EO-S12#pre-trained-models)?

Thanks a lot again.

**Comment by wangyi111** [comment]
Hi @Zhou2019, thanks for your question!

> What is the RGB version? What data does it contain?

The RGB version includes 3-band sentinel-2 L1C data (compressed to uint8) of the whole dataset, i.e., full 250k locations each with 4 seasons. The data size is very much compressed so convenient to use if you only need RGB bands.

> Is it convenient for the author to provide the performance metrices of the model in the second table of the [Pre-trained models](https://github.com/zhu-xlab/SSL4EO-S12#pre-trained-models)?

What exactly are the models / backbones / input data / downstream datasets you want? For that table, we haven't conducted the experiments on all datasets.

**Comment by wangyi111** [comment]
We don't have them currently but they are within further plans. Stay tuned!

**Comment by wangyi111** [comment]
No problem, thanks for your interest!

> the data format of Compressed dataset is not LMDB

Yes, for the published data we provide common GeoTIFF files, and it depends on the users how to use them. Loading the data from individual files is a common way, and we have `ssl4eo_dataset.py` for reference, which would create a Pytorch dataset class and you can directly use this class in the pre-training scripts.
 
In our experiments, we converted the data to a single LMDB file with the function `make_lmdb()` because of two reasons:
1. Our server has enough storage but has a limit of INODE (the number of files), thus the raw data can't fit in (~14M files). After conversion to LMDB, it's only a single file.
2. Data loading from LMDB is faster than from a lot of individual image files.

> the ssl4eo_dataset.py file provided with the codebase does not support such conversion

Check the [`make_lmdb()`](https://github.com/zhu-xlab/SSL4EO-S12/blob/8b943466396c1eedb031f7175db0a0df167d0bc0/src/benchmark/pretrain_ssl/datasets/SSL4EO/ssl4eo_dataset.py#L183) function to convert data to LMDB. You may need to modify it for your need.

> Specifically, I plan to modify the ssl4eo_dataset.py file by changing line 92 to bands = RGB_BANDS.
And use the S2C dataset in the compressed dataset, which means directly selecting three bands from S2C and converting them into an RGB dataset in LMDB format.

If you only need RGB data, you can directly download the [RGB version](https://mediatum.ub.tum.de/1702379). And yes you need to modify `ssl4eo_dataset.py`, especially `make_lmdb()` function (you only have s2c rgb data). In this case, the RGB data is already normalized to [0,255], so you don't need `normalize`. 

> And the argument of normalize should be set to True? Or any other argument that require special configuration.

The "normalize" normalizes the data from uint16 to [0,255] with band-wise mean and std. If you use the full s2c dataset, it is optional. If you use the RGB version, it is already normalized so no need.

> Alternatively, would it be possible for you to share an RGB version of the LMDB format dataset?

Unfortunately the LMDB files are very big (e.g. 1TB for s2c), so we are not able to share them. Rather we can help you create it with the `make_lmdb()` function. It should not be too complicated. Let us know if you have any questions.

P.S. I have to admit the current codes lack documentation, we'll work on it:)

**Comment by wangyi111** [comment]
Just back from vacation, sorry for the late reply. This is DINO pretraining right? Are you using mixed precision (`args.use_fp16`)? If so you can try turning it off. 

This issue was reported in the original DINO repo as well, you can check it here https://github.com/facebookresearch/dino/issues/58.

**Comment by wangyi111** [comment]
> case 1: loss is decrease very slowly, almost not decreasing.
Does this work for your downstream task? E.g. better than random when doing linear classification.

> Is there a problem with the RGB version dataset I synthesized? I found that the RGB lmdb format dataset file I synthesized is 289G, much larger than the original file.

it is normal to become larger when converting compressed files to binary lmdb file. But you may want to check that your training data is good by e.g. visualizing some patches using your dataloaders , etc. Another thing is that our provided RGB version is JPEG compressed which is more lossy compared to the raw data, but I don't expect this to be a big issue.

> Is DINO not suitable for an RGB version dataset?

We didn't try DINO on RGB, so can't tell the answer unfortunately. But from what I expected it shouldn't be too bad. When I have time I can try running one, but I don't have any idea about any special hyperparameter settings so far. Also, MoCo should be possible to work on a single multi-GPU machine, and yes some modifications may be needed. It also gives a rather robust training (not very sensitive to parameters).

As you are using RGB only, maybe you can try loading an SSL pre-trained model from ImageNet?
- From our experience, vision models transfer very well to RGB tasks in RS (comparable to training our RS data). 
- You can maybe also try initializing your model with some ImageNet pre-trained weights (like DINO models), then use a small learning rate to fine-tune pre-training from that.

Training SSL can be tricky:) Hope you will solve your training issues soon.

**Comment by d3tk** [comment]
@Zhou2019 Hello, I am trying to train a ViT-s with DINO on the RGB dataset. Did you have any success with this? If so, would you care to share the weights or the code/modifications you made?
Thank you!
